x-pack/plugins/fleet/server/services/agent_policies/full_agent_policy.ts

```diff
@@ -0,0 +1,228 @@
+  } catch (err) {
+    if (!err.isBoom || err.output.statusCode !== 404) {
+      throw err;
+    }
+  }
+
+  if (!agentPolicy) {
+    return null;
+  }
+
+  const defaultOutputId = await outputService.getDefaultOutputId(soClient);
+  if (!defaultOutputId) {
+    throw new Error('Default output is not setup');
+  }
+
+  const dataOutputId = agentPolicy.data_output_id || defaultOutputId;
+  const monitoringOutputId = agentPolicy.monitoring_output_id || defaultOutputId;
+
+  const outputs = await Promise.all(
+    Array.from(new Set([dataOutputId, monitoringOutputId])).map((outputId) =>
+      outputService.get(soClient, outputId)
+    )
+  );
+
+  const dataOutput = outputs.find((output) => output.id === dataOutputId);
+  if (!dataOutput) {
+    throw new Error(`Data output not found ${dataOutputId}`);
+  }
+  const monitoringOutput = outputs.find((output) => output.id === monitoringOutputId);
+  if (!monitoringOutput) {
+    throw new Error(`Monitoring output not found ${monitoringOutputId}`);
+  }
+
+  const fullAgentPolicy: FullAgentPolicy = {
+    id: agentPolicy.id,
+    outputs: {
+      ...outputs.reduce<FullAgentPolicy['outputs']>((acc, output) => {
+        acc[getOutputIdForAgentPolicy(output)] = transformOutputToFullPolicyOutput(output);
+
+        return acc;
+      }, {}),
+    },
+    inputs: storedPackagePoliciesToAgentInputs(
+      agentPolicy.package_policies as PackagePolicy[],
+      getOutputIdForAgentPolicy(dataOutput)
+    ),
+    revision: agentPolicy.revision,
+    ...(agentPolicy.monitoring_enabled && agentPolicy.monitoring_enabled.length > 0
+      ? {
+          agent: {
+            monitoring: {
+              namespace: agentPolicy.namespace,
+              use_output: getOutputIdForAgentPolicy(monitoringOutput),
+              enabled: true,
+              logs: agentPolicy.monitoring_enabled.includes(dataTypes.Logs),
+              metrics: agentPolicy.monitoring_enabled.includes(dataTypes.Metrics),
+            },
+          },
+        }
+      : {
+          agent: {
+            monitoring: { enabled: false, logs: false, metrics: false },
+          },
+        }),
+  };
+
+  const dataPermissions = (await storedPackagePoliciesToAgentPermissions(
+    soClient,
+    agentPolicy.package_policies
+  )) || { _fallback: DEFAULT_PERMISSIONS };
+
+  dataPermissions._elastic_agent_checks = {
+    cluster: DEFAULT_PERMISSIONS.cluster,
+  };
+
+  // TODO: fetch this from the elastic agent package
+  const monitoringNamespace = fullAgentPolicy.agent?.monitoring.namespace;
+  const monitoringPermissions: FullAgentPolicyOutputPermissions =
+    monitoringOutputId === dataOutputId
+      ? dataPermissions
+      : {
+          _elastic_agent_checks: {
+            cluster: DEFAULT_PERMISSIONS.cluster,
+          },
+        };
+  if (
+    fullAgentPolicy.agent?.monitoring.enabled &&
+    monitoringNamespace &&
+    monitoringOutput &&
+    monitoringOutput.type === 'elasticsearch'
```

Comment:
Should this string `"elasticsearch"` move to a constant e.g. `MONITORING_OUTPUT_TYPE_ELASTICSEARCH`?